src/components/types.ts

```diff
@@ -151,3 +151,69 @@ export interface VideoFormat extends MediaFormat {
   /** URL of an image to be shown while the video is downloading. If not provided, nothing is displayed until the first frame is available. */
   poster?: string
 }
+
+export interface BaseInputProps {
+  /** Id of the current user. */
+  sender: string
+  /** Id of the current conversation. */
+  conversationId: string
+  /** Label text to be displayed in the input, which will then move to the top when the input is focused on. If both label and placeholder are provided, the placeholder will only be visible once the input is focused on. */
+  label?: string
+  /** Placeholder text to be displayed in the input before user starts typing. */
+  placeholder?: string
+  /** Boolean that dictates whether `TextInput` can expand to be multiline. */
+  multiline?: boolean
+  /** Maximum number of rows to be displayed. */
+  maxRows?: number
+  /** Boolean that dictates whether `TextInput` takes up 100% width of the parent container. */
+  fullWidth?: boolean
+  /** function to send the message */
+  send: (message: Message) => void
+  /** Additional condition to enable the send button. */
+  isSendEnabled?: boolean
+  /** Boolean to enable speech-to-text. See which browsers are supported [here](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility). */
+  enableSpeechToText?: boolean
+}
+
+export interface TextInputProps
+  extends Omit<BaseInputProps, 'send' | 'isSendEnabled'> {
+  ws: WebSocketClient
+}
+
+export type MultimodalInputProps = TextInputProps &
+  Omit<
+    UploaderProps,
+    | 'messageId'
+    | 'handleFileCountChange'
+    | 'filePreviewRef'
+    | 'errorMessagesRef'
+  >
+
+export interface UploaderProps {
```

Comment:
This is declared after it is being used. Please re-order such that declaration is before any usage.